deepchem/molnet/load_function/material_datasets/load_Pt_NO_surface_adsorbate_energy.py

```diff
@@ -0,0 +1,115 @@
+"""
+Platinum Adsorbtion structure for N and NO along with their formation energies
+"""
+import numpy as np
+import os
+import deepchem as dc
+from deepchem.molnet.load_function.molnet_loader import TransformerGenerator, _MolnetLoader
+from deepchem.feat.material_featurizers.lcnn_featurizer import LCNNFeaturizer
+from deepchem.data import Dataset
+from typing import List, Optional, Tuple, Union
+
+PLATINUM_URL = "https://deepchemdata.s3-us-west-1.amazonaws.com/datasets/platinum_adsorption.tar.gz"
+PLATINUM_TASKS = ["Formation Energy"]
+PRIMITIVE_CELL_TEMPLATE = """#Primitive Cell
+2.81852800e+00  0.00000000e+00  0.00000000e+00 T
+-1.40926400e+00  2.44091700e+00  0.00000000e+00 T
+0.00000000e+00  0.00000000e+00  2.55082550e+01 F
+1 1
+1 0 2
+6
+0.666670000000  0.333330000000  0.090220999986 S1
+0.333330000000  0.666670000000  0.180439359180 S1
+0.000000000000  0.000000000000  0.270657718374 S1
+0.666670000000  0.333330000000  0.360876077568 S1
+0.333330000000  0.666670000000  0.451094436762 S1
+0.000000000000  0.000000000000  0.496569911270 A1"""
+
+
+class _PtAdsorptionLoader(_MolnetLoader):
+
+  def create_dataset(self) -> Dataset:
+    dataset_file = os.path.join(self.data_dir, 'Platinum_Adsorption.json')
+    if not os.path.exists(dataset_file):
+      dc.utils.data_utils.download_url(url=PLATINUM_URL, dest_dir=self.data_dir)
+      dc.utils.data_utils.untargz_file(
+          os.path.join(self.data_dir, 'platinum_adsorption.tar.gz'),
+          self.data_dir)
+    loader = dc.data.JsonLoader(
+        tasks=PLATINUM_TASKS,
+        feature_field="Structures",
+        label_field="Formation Energy",
+        featurizer=self.featurizer,
+        **self.args)
+    return loader.create_dataset(dataset_file)
+
+
+def load_Platinum_Adsorption(
+    featurizer: Union[dc.feat.Featurizer, str] = LCNNFeaturizer(
+        np.around(6.00), PRIMITIVE_CELL_TEMPLATE),
+    splitter: Union[dc.splits.Splitter, str, None] = 'random',
```

Comment:
Should this be `'scaffold'`?  We're trying to standardize on scaffold splitting wherever it makes sense.

Reply:
As I just went through the code for _generate_scaffold , it uses SMILE strings. But in this case , we are don't have chemical structures. Its just the coordinates of Adsorbate molecules (i.e N and NO ) for a platinum lattice. The input data looks in this format
Ex- 
2.81859800e+00  0.00000000e+00  0.00000000e+00
    -1.40929900e+00  2.44097800e+00  0.00000000e+00
     0.00000000e+00  0.00000000e+00  2.55082550e+01
    6
     0.000000000000  0.000000000000  0.090220999986 S1
     0.500000499894  0.622008360788  0.180442000011 S1
     0.999999500106  0.666666711253  0.270892474701 S1
     0.000000000000  0.000000000000  0.361755713893 S1
     0.500000499894  0.622008360788  0.454395429618 S1
     0.000000000000  0.666667212896  0.502346789304 A1 1